### PR TITLE
Add controller specs

### PR DESCRIPTION
**Pull request by SamTush**
## What does this PR do?

This PR is raised to implement the following features: 👍 :

- [x] Add RSpec as a gem
- [x] Create a Request spec file

## Description of the task to be completed ↓↓ 

-  Controllers specs for the app.

## How to test

- Run

```
   bundle exec rspec
```

Screenshot 2023-06-23 at 15 51 28

#### Thanks for reviewing my PR 😊

### CODE REVIEW

File: spec/requests/posts_controller_specs.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      get user_posts_path(user)
+      expect(response.body).to include('Here is a list of posts for a given user')
+    end
+  end
+
+  describe 'GET /users/:user_id/posts/:id' do
+    let(:user) { User.create(name: 'Stella', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Rusia.', posts_counter: 0) }
+    let(:post) do
+      Post.create(author: user, title: 'Sample Post', text: 'My first post', comments_counter: 0, likes_counter: 0)
+      # Post.create(author: user, title: 'Hello4', text: 'This is my fourth post')
```
Comment (AnselemOdims) [inline spec/requests/posts_controller_specs.rb]: - [ ] Good job writing the tests but please kindly remove this commented code from your codebase to keep it clean.

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @SamTush , 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by SamTush** [comment]
Thank you @AnselemOdims  for reviewing my PR👍🏽 cheers and happy coding😎

**Review by alexander16108 — APPROVED** [review]
Hi @SamTush , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

✅ Correct use of Flow
✅ Descriptive Pull Request

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**